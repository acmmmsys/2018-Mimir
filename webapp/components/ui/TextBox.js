import React, { Component } from 'react';
import classnames from 'classnames';
import { compose, withState, withHandlers } from 'recompose';

import 'style/ui/TextBox.scss';

const defaultClassNames = [
    'text-boxcontainer',
    'input-group',
];

class TextBox extends Component {
    
    static defaultProps = {
        classNames: [],
        className: '',
    }

    shouldComponentUpdate(nextProps, nextState) {
        return this.props.value !== nextProps.value;
    }

    render() {

        let classNames = classnames(
            defaultClassNames,
            this.props.className,
        );

        return (
            <div className={classNames}>
                <label style={{display: 'block', textAlign: 'left'}}>
                    {this.props.label}
                </label>
                <div className={'text-box'} 
                    contentEditable={true}
                    style={{width: '100%'}}
                    onInput={this.props.onChange}
                >
                    {this.props.value}
                </div>
            </div>
        )
    }
}

export default TextBox;