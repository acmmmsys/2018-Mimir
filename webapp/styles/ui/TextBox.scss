
.text-box-container {

    label {
        display: block;
        text-align: left;
        width: 100%;
    }

    .text-box {
        border: none;
        border: solid 2px white;
        color: black;

        &:focus {
            outline: none;
            border: solid 2px #c9c9c9;   
        }
    }
}