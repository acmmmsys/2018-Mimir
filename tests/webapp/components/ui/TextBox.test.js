import React from 'react';
import TextBox from '../../../../webapp/components/ui/TextBox'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
 
const mockStore = configureStore();

let store;
let wrapper;

test('TextBox does render', () => {
  const wrapper = shallow(
      <TextBox store={mockStore(initialState)}>This is a test!</TextBox>
  );
  expect(wrapper).toMatchSnapshot();
});
