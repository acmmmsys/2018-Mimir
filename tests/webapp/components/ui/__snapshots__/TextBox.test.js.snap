// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox does render 1`] = `
<div
  className="text-boxcontainer input-group"
>
  <label
    style={
      Object {
        "display": "block",
        "textAlign": "left",
      }
    }
  />
  <div
    className="text-box"
    contentEditable={true}
    style={
      Object {
        "width": "100%",
      }
    }
  />
</div>
`;
